### PR TITLE
internal/cmd: more consolidation and use of the operator.Configuration type

### PR DESCRIPTION
**Pull request by joelanford**
**Description of the change:**

Internal refactoring to make more re-use of existing code.

- Use `operator.Configuration` in scorecard.
- Remove duplicative scorecard-specific client config helpers.
- Move `operator.Configuration` load call for the `run` command to the root of the run command so it can be reused by both `packagemanifests` and `bundle`.

**Motivation for the change:**
Reduce code duplication

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] ~Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))~
- [ ] ~Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)~

### CODE REVIEW

**Comment by jberkhahn** [comment]
So, not to look a gift horse in the mouth or anything, but if it is a goal to eventually have all our tests in ginkgo/gomega (which it is, i think?), why are we still adding raw gotest tests?

**Comment by joelanford** [comment]
@jberkhahn These tests existed already in for the scorecard code that was removed. So I moved those tests over and refactored them a little to work for `config.Load`.

File: internal/cmd/operator-sdk/scorecard/cmd.go [hunk 1]
```diff
@@ -28,50 +28,55 @@ import (
...
 		// to run it, etc.
 		Long: `Has flags to configure dsl, bundle, and selector. This command takes
 one argument, either a bundle image or directory containing manifests and metadata.
 If the argument holds an image tag, it must be present remotely.`,
 		PreRunE: func(cmd *cobra.Command, args []string) (err error) {
-			return c.validate(args)
+			if err := c.validate(args); err != nil {
+				return err
+			}
+			return c.cfg.Load()
```
Comment (camilamacedo86) [inline internal/cmd/operator-sdk/scorecard/cmd.go:69]: 👍

File: internal/cmd/operator-sdk/scorecard/cmd.go [hunk 2]
```diff
@@ -152,19 +157,19 @@ func (c *scorecardCmd) run() (err error) {
...
-		runner := scorecard.PodTestRunner{
-			ServiceAccount: c.serviceAccount,
-			Namespace:      scorecard.GetKubeNamespace(c.kubeconfig, c.namespace),
-			BundlePath:     c.bundle,
-			BundleMetadata: metadata,
-		}
-
 		// Only get the client if running tests.
-		if runner.Client, err = scorecard.GetKubeClient(c.kubeconfig); err != nil {
+		kubeClient, err := kubernetes.NewForConfig(c.cfg.RESTConfig)
```
Comment (camilamacedo86) [inline internal/cmd/operator-sdk/scorecard/cmd.go:161]: 👍

**Review by camilamacedo86 — APPROVED** [review]
It shows great for me 👍 
/lgtm

**Comment by camilamacedo86** [comment]
Hi @joelanford, 

Let's rebase it with the master and get it merged?

**Review by camilamacedo86 — COMMENTED** [review]
@joelanford 

We will lose this cleanup. Is it no longer required?

**Pull request closed without merging**